save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityTableView.kt

```diff
@@ -0,0 +1,354 @@
+/**
+ * View for vulnerability tables
+ */
+
+package com.saveourtool.save.frontend.components.views.vuln
+
+import com.saveourtool.save.entities.cosv.VulnerabilityMetadataDto
+import com.saveourtool.save.filters.VulnerabilityFilter
+import com.saveourtool.save.frontend.TabMenuBar
+import com.saveourtool.save.frontend.components.basic.renderOrganizationWithName
+import com.saveourtool.save.frontend.components.basic.renderUserAvatarWithName
+import com.saveourtool.save.frontend.components.basic.table.filters.vulnerabilitiesFiltersRow
+import com.saveourtool.save.frontend.components.tables.*
+import com.saveourtool.save.frontend.components.views.contests.tab
+import com.saveourtool.save.frontend.externals.i18next.useTranslation
+import com.saveourtool.save.frontend.themes.Colors
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.noopResponseHandler
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.validation.FrontendRoutes
+
+import js.core.jso
+import react.*
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.td
+import react.dom.html.ReactHTML.th
+import react.dom.html.ReactHTML.tr
+import react.router.dom.Link
+import web.cssom.Border
+import web.cssom.ClassName
+import web.cssom.ColorProperty
+import web.cssom.rem
+
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * [FC] for [vulnerabilityTableView]
+ */
+val vulnerabilityTableView: FC<VulnerabilityTableViewProps> = FC { props ->
+    val (t, i18n) = useTranslation(TABLE_HEADERS_LOCALE_NAMESPACE)
+
+    val (isVulnerabilityCollectionPage, _) = useState(props.currentUserInfo != null && props.userName == null && props.organizationName == null)
+    val (isUserProfilePage, _) = useState(props.currentUserInfo == null && props.userName != null && props.organizationName == null)
+    val (isOrganizationPage, _) = useState(props.currentUserInfo == null && props.userName == null && props.organizationName != null)
+
+    val (vulnerabilityFilter, setVulnerabilityFilter) = useState(
+        when {
+            isVulnerabilityCollectionPage -> props.filters ?: VulnerabilityFilter.approved
+            isUserProfilePage -> VulnerabilityFilter(authorName = props.userName)
+            isOrganizationPage -> VulnerabilityFilter(organizationName = props.organizationName)
+            else -> VulnerabilityFilter.empty
+        }
+    )
+    val (countVulnerability, setCountVulnerability) = useState(0)
+
+    val (selectedMenu, setSelectedMenu) = useState(VulnerabilityListTab.PUBLIC)
+    val (isOwnerTab, setIsOwnerTab) = useState(false)
```

Comment:
can be replaced by checking `selectedMenu`

Reply:
Done